src/components/Store.js

```diff
@@ -0,0 +1,80 @@
+import { useState } from 'react'
```

Comment:
this store looks hella complex and i dont undersatnd most of it but i really like how neat and tidy the code is